packages/replay-next/src/suspense/createGenericCache.ts

```diff
@@ -13,18 +13,24 @@ import {
   Wakeable,
 } from "./types";
 
+export { STATUS_PENDING, STATUS_REJECTED, STATUS_RESOLVED } from "./types";
+
+type CacheRecordStatus = typeof STATUS_PENDING | typeof STATUS_RESOLVED | typeof STATUS_REJECTED;
+
 export interface GenericCache<TParams extends Array<any>, TValue> {
   getValueSuspense(...args: TParams): TValue;
   getValueAsync(...args: TParams): Thennable<TValue> | TValue;
   getValueIfCached(...args: TParams): { value: TValue } | undefined;
   addValue(value: TValue, ...args: TParams): void;
+  getStatus(...args: TParams): CacheRecordStatus | undefined;
```

Comment:
I wonder if `STATUS_UNINITIALIZED` would be a bit clearer?

My personal preference is to avoid `undefined` since it can mask unhandled edge cases.

Reply:
Hmm. We don't have a `STATUS_UNINITIALIZED` value atm.

Reply:
I know. I was suggesting we _add one_ rather than `undefined` because `undefined` seems like a bad pattern since it can mask unhandled edge cases.